src/components/form/form.styles.ts

```diff
@@ -6,19 +6,161 @@
 
 export const euiFormVariables = (euiThemeContext: UseEuiTheme) => {
-  const { euiTheme } = euiThemeContext;
+  const { euiTheme, colorMode } = euiThemeContext;
+  const isColorDark = colorMode === 'DARK';
+  const backgroundColor = isColorDark
+    ? shade(euiTheme.colors.lightestShade, 0.4)
+    : tint(euiTheme.colors.lightestShade, 0.6);
+
+  const controlHeight = euiTheme.size.xxl;
+  const controlCompressedHeight = euiTheme.size.xl;
 
   return {
+    // Sizing
     maxWidth: `${euiTheme.base * 25}px`,
-    controlHeight: euiTheme.size.xxl,
-    controlCompressedHeight: euiTheme.size.xl,
+    controlHeight: controlHeight,
+    controlCompressedHeight: controlCompressedHeight,
     controlPadding: euiTheme.size.m,
     controlCompressedPadding: euiTheme.size.s,
     controlBorderRadius: euiTheme.border.radius.medium,
     controlCompressedBorderRadius: euiTheme.border.radius.small,
-    checkboxBorderRadius: euiTheme.border.radius.small,
+
+    // Coloring
+    backgroundColor: backgroundColor,
+    backgroundDisabledColor: darken(euiTheme.colors.lightestShade, 0.1),
+    backgroundReadOnlyColor: euiTheme.colors.emptyShade,
+    borderColor: transparentize(euiTheme.border.color, 0.9),
+    borderDisabledColor: transparentize(euiTheme.border.color, 0.9),
+    controlDisabledColor: euiTheme.colors.mediumShade,
+    controlBoxShadow: '0 0 transparent',
+    controlPlaceholderText: makeHighContrastColor(euiTheme.colors.subduedText)(
+      backgroundColor
+    ),
+    inputGroupLabelBackground: isColorDark
+      ? shade(euiTheme.colors.lightShade, 0.15)
+      : tint(euiTheme.colors.lightShade, 0.5),
+    inputGroupBorder: 'none',
+
+    // Coloring - specific for checkboxes and radios
+    customControlDisabledIconColor: isColorDark
+      ? shade(euiTheme.colors.mediumShade, 0.38)
+      : tint(euiTheme.colors.mediumShade, 0.48),
+    customControlBorderColor: isColorDark
+      ? shade(euiTheme.colors.lightestShade, 0.18)
+      : tint(euiTheme.colors.lightestShade, 0.38),
+
+    // Icon sizes
+    controlIconSize: {
+      s: euiTheme.size.m,
+      m: euiTheme.size.base,
+      l: euiTheme.size.l,
+      xl: euiTheme.size.xl,
+      xxl: euiTheme.size.xxl,
+    },
+
+    // Control layout
+    controlLayoutGroupInputHeight: mathWithUnits(controlHeight, (x) => x - 2),
+    controlLayoutGroupInputCompressedHeight: mathWithUnits(
+      controlCompressedHeight,
+      (x) => x - 2
+    ),
+    controlLayoutGroupInputCompressedBorderRadius: euiTheme.border.radius.small,
   };
 };
+
+export const euiFormCustomControlBorderColor = ({
+  euiTheme,
+  colorMode,
+}: UseEuiTheme) => `
+  ${
+    colorMode === 'DARK'
+      ? shade(euiTheme.colors.lightestShade, 0.82)
+      : tint(euiTheme.colors.lightestShade, 0.7)
+  }
+`;
+
+export const euiFormControlSize = ({
+  euiThemeContext,
+  height,
+  includeAlternates,
+}: {
+  euiThemeContext: UseEuiTheme;
+  height?: string;
+  includeAlternates?: 'fullWidth' | 'compressed' | 'inGroup';
+}) => {
```

Comment:
Based on our other Emotion utils, `euiThemeContext` should always come as the first arg

```suggestion
export const euiFormControlSize = (
  euiThemeContext: UseEuiTheme,
  options: {
    height?: string;
    includeAlternates?: 'fullWidth' | 'compressed' | 'inGroup';
  }
) => {
```